testutil/testdata/proto.proto

```diff
@@ -70,3 +70,274 @@ message BadMultiSignature {
   repeated bytes signatures = 1;
   bytes malicious_field = 5;
 }
+
+message Customer1 {
+    int32 id = 1;
+    string name = 2;
+    float subscription_fee = 3;
+
+    string payment = 7;
+}
+
+message Customer2 {
+    int32 id = 1;
+    int32 industry = 2;
+    string name = 3;
+    float fewer = 4;
+
+    int64 reserved = 1047;
+
+    enum City {
+        Laos = 0;
+        LosAngeles = 1;
+        PaloAlto = 2;
+        Moscow = 3;
+        Nairobi = 4;
+    }
+
+    City city = 6;
+
+    google.protobuf.Any miscellaneous = 10;
+}
+
+message Nested4A {
+    int32 id = 1;
+    string name = 2;
+}
+
+message Nested3A {
+    int32 id = 1;
+    string name = 2;
+    repeated Nested4A a4 = 4;
+    map<int64, Nested4A> index = 5;
+}
+
+message Nested2A {
+    int32 id = 1;
+    string name = 2;
+    Nested3A nested = 3;
+}
+
+message Nested1A {
+    int32 id = 1;
+    Nested2A nested = 2;
+}
+
+message Nested4B {
+    int32 id = 1;
+    int32 age = 2;
+    string name = 3;
+}
+
+message Nested3B {
+    int32 id = 1;
+    int32 age = 2;
+    string name = 3;
+    repeated Nested4B b4 = 4;
+}
+
+message Nested2B {
+    int32 id = 1;
+    double fee = 2;
+    Nested3B nested = 3;
+    string route = 4;
+}
+
+message Nested1B {
+    int32 id = 1;
+    Nested2B nested = 2;
+    int32 age = 3;
+}
+
+message Customer3 {
+    int32 id = 1;
+    string name = 2;
+    float sf = 3;
+    float surcharge = 4;
+    string destination = 5;
+
+    oneof payment {
+        string credit_card_no = 7;
+        string cheque_no = 8;
+    }
+
+    Customer1 original = 9;
+}
+
+message TestVersion1 {
+    int64 x = 1;
+    TestVersion1 a = 2;
+    TestVersion1 b = 3; // [(gogoproto.nullable) = false] generates invalid recursive structs;
+    repeated TestVersion1 c = 4;
+    repeated TestVersion1 d = 5 [(gogoproto.nullable) = false];
+    oneof sum {
+       int32 e = 6;
+       TestVersion1 f = 7;
+    }
+    google.protobuf.Any g = 8;
+    repeated TestVersion1 h = 9; // [(gogoproto.castrepeated) = "TestVersion1"];
+    // google.protobuf.Timestamp i = 10;
+    // google.protobuf.Timestamp j = 11; // [(gogoproto.stdtime) = true];
+    Customer1 k = 12 [(gogoproto.embed) = true];
```

Comment:
Can we perhaps do this cleanup in a follow-on PR? I ask because in our discussion we wanted some of these fields for test cases but they couldn't work out quite as planned. Important to keep them first so they don't get lost.

Reply:
Yeah, let's address in a follow-up PR. I'm adding a note in #6192